### ISSUE TITLE
Implement 'persist_chunk' in TrainingDataChunk

### ISSUE THREAD

**Issue by tabergma** [RasaHQ/rasa#7276]
related to #6836

In order to do partial dataset training we need to write down the featurized chunks. Implement the method `persist_chunk` in `TrainingDataChunk`. The method should write all examples, e.g. labels and features, of the chunk to a [TFRecord](https://www.tensorflow.org/tutorials/load_data/tfrecord#writing_a_tfrecord_file) file. You need to convert each `Message` to a [tf.train.Example](https://tensorflow.google.cn/tutorials/load_data/tfrecord#tftrainexample) and afterwards [write](https://tensorflow.google.cn/tutorials/load_data/tfrecord#tfrecord_files_in_python) them down.

Note: We need to figure out how to handle sparse features.

**Note**: This feature should branch off from `partial_dataset_loading` and should be squashed and merged back into that branch.
